### ISSUE TITLE
Tests against a set of production pages hang and die after running 8-12 tests in Node 8 & 10 but not 11

### ISSUE THREAD

**Issue by jcollum-nike** [DevExpress/testcafe#3483]
### What is your Test Scenario?

I want to hit about 150 production web pages with testcafe to verify basic functionality

### What is the Current behavior?

The tests run fine if they are run against Node 11 runtime but fail after running about 10 tests in Node 8 or Node 10. Please note: I can run all ~150 tests when running in Node 11 but can only run about 10 tests before the browser dies when running in Node 8 or 10. Also they fail the same way with or without the proxy (tested against work network with proxy and non-work network on mobile hotspot).

Also worth noting that the tests run fine in Node 8 & Cypress.    

### What is the Expected behavior?

All tests run successfully.

### What is your web application and your TestCafe test code?

Your website URL (or attach your complete example): `https://nike.com/ru/help/a/подлинность-товаров-nike` is a typical link but they vary a lot, see test output below

<details>
<summary>Your complete test code (or attach your test files):</summary>

```js
 import { Selector } from 'testcafe';

 // this is really just a set of languages and url portions (e.g. ru and подлинность-товаров-nike)
 // the full urls are in the test output below
 import slugSettings from './slugs';

 Object.keys(slugSettings).forEach(k => {
   const setting = slugSettings[k];
   const baseUrl = 'https://nike.com';
   const fixtureName = `Production Article Test - ${setting.description}`;

   fixture`${fixtureName}`.page`${baseUrl}/${setting.country}/help`;

   setting.slugs.forEach(slug => {
     const articleUrl = `${baseUrl}/${setting.country}/help/a/${slug}`;
     test.page(`${articleUrl}`)(`Slug: ${slug} `, async t => {
       const h1Select = 'div.article-content h1';
       console.log(`  Testing ${slug}, url: ${articleUrl}`);

       //await t.debug();

       // check to see if the error box is present
       if (await Selector('div.error-box h1').exists) {
         throw new Error(`Failed to load page at ${articleUrl}`);
       }

       await t
         .expect(Selector(h1Select), { visibilityCheck: true })
         .ok(`First H1 in article was not found at url ${articleUrl}`);

       const h1Value = await Selector(h1Select).innerText;

       await t
         .expect(h1Value.length)
         .gt(6, 'First H1 in article did not have required string length', { timeout: 1000 });
     });
   });
 });
```
</details>

<details>
<summary>Your complete test report:</summary>

```
 Production Article Test - Russia
  Testing подлинность-товаров-nike, url: https://nike.com/ru/help/a/подлинность-товаров-nike
 ✓ Slug: подлинность-товаров-nike
  Testing отзывы-о-товарах, url: https://nike.com/ru/help/a/отзывы-о-товарах
 ✓ Slug: отзывы-о-товарах
  Testing план-тренировок-nrc, url: https://nike.com/ru/help/a/план-тренировок-nrc
 ✓ Slug: план-тренировок-nrc
  Testing настройки-nrc, url: https://nike.com/ru/help/a/настройки-nrc
 ✓ Slug: настройки-nrc
  Testing связать-nrc-ntc, url: https://nike.com/ru/help/a/связать-nrc-ntc
 ✓ Slug: связать-nrc-ntc

 Production Article Test - Thailand
  Testing รองเท้า-สำหรับ-ลดการบิดเท้าออกด้านนอก, url: https://nike.com/th/help/a/รองเท้า-สำหรับ-ลดการบิดเท้าออกด้านนอก
 ✓ Slug: รองเท้า-สำหรับ-ลดการบิดเท้าออกด้านนอก
  Testing ยกเลิกการสมัคร-จดหมายข่าว, url: https://nike.com/th/help/a/ยกเลิกการสมัคร-จดหมายข่าว
 ✓ Slug: ยกเลิกการสมัคร-จดหมายข่าว
  Testing การดูแลรักษาสินค้า-drifit, url: https://nike.com/th/help/a/การดูแลรักษาสินค้า-drifit
 ✓ Slug: การดูแลรักษาสินค้า-drifit

 Production Article Test - Japan
  Testing nrc-ntc-アプリ, url: https://nike.com/jp/help/a/nrc-ntc-アプリ
 ✓ Slug: nrc-ntc-アプリ
  Testing ラン-nrc, url: https://nike.com/jp/help/a/ラン-nrc
 ✓ Slug: ラン-nrc
  Testing nrcとパートナーアプリ-デバイスの接続, url: https://nike.com/jp/help/a/nrcとパートナーアプリ-デバイスの接続
 ✓ Slug: nrcとパートナーアプリ-デバイスの接続

 Production Article Test - America
  Testing free-shipping, url: https://nike.com/us/help/a/free-shipping
 ✓ Slug: free-shipping
  Testing shipping-delivery, url: https://nike.com/us/help/a/shipping-delivery
ERROR The HeadlessChrome 72.0.3626 / Mac OS X 10.13.6 browser disconnected. This problem may appear when a browser hangs or is closed, or due to network issues.
```
</details>

### Steps to Reproduce:

Run the test with the command line below.

### Your Environment details:

* testcafe version:                  1.0.1
* node.js version:                    v8.15.0
* command-line arguments:  `testcafe chrome:headless --proxy 165.225.50.127  --skip-js-errors --screenshots-on-fails --screenshots ./screenshots --screenshot-path-pattern '${TEST}.png' ./tests/article/prodArticles.test.js`
* browser name and version: Chrome 72.0.3626
* platform and version:          Node v8.15.0
   NPM 6.4.1
   Yarn 1.12.3
   OS:
   ProductName:	Mac OS X
   ProductVersion:	10.13.6
   BuildVersion:	17G5019
* other:

**Comment by AlexKamaev** [comment]
I was able to reproduce the issue using the following code:

```JS
import { Selector } from 'testcafe';

// this is really just a set of languages and url portions (e.g. ru and подлинность-товаров-nike)
// the full urls are in the test output below
const slugSettings = [{
    country:     'ru',
    description: 'nike',
    slugs:       ['подлинность-товаров-nike']
}];

Object.keys(slugSettings).forEach(k => {
    const setting     = slugSettings[k];
    const baseUrl     = 'https://nike.com';
    const fixtureName = `Production Article Test - ${setting.description}`;

    fixture`${fixtureName}`.page`${baseUrl}/${setting.country}/help`;

    for (let i = 0; i < 10; i++) {
        setting.slugs.forEach(slug => {
            const articleUrl = `${baseUrl}/${setting.country}/help/a/${slug}`;
            test.page(`${articleUrl}`)(`Slug: ${slug} `, async t => {
                const h1Select = 'div.article-content h1';
                console.log(`  Testing ${slug}, url: ${articleUrl}`);

                //await t.debug();

                // check to see if the error box is present
                if (await Selector('div.error-box h1').exists) {
                    throw new Error(`Failed to load page at ${articleUrl}`);
                }

                await t
                    .expect(Selector(h1Select), { visibilityCheck: true })
                    .ok(`First H1 in article was not found at url ${articleUrl}`);

                const h1Value = await Selector(h1Select).innerText;

                await t
                    .expect(h1Value.length)
                    .gt(6, 'First H1 in article did not have required string length', { timeout: 1000 });
            });
        });
    }
});
```
The cause of the issue appears to be in https://github.com/DevExpress/testcafe-hammerhead/issues/1885
Please refer to the following [comment ](https://github.com/DevExpress/testcafe/issues/3438#issuecomment-466368768) to get a workaround. I've checked it on the code above and it works.

**Comment by jcollum-nike** [comment]
Sorry for the double post, I had planned on deleting this issue because docker isn't the issue. I didn't see that it had gotten a response.

**Comment by AndreyBelym** [comment]
Fixed in `testcafe-1.0.0-rc.1`.
